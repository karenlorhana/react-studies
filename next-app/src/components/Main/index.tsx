import React from 'react'
import { Head } from 'next/document'
import * as S from './styles'

const Main = () => (
  <S.Wrapper>
    <S.Logo src="/img/react-image.png" alt="Logo do react" />
    <S.Title> React Avan√ßado</S.Title>
    <S.Description>
      Typescript, ReactJS, NextJS and Styled Components
    </S.Description>
    <S.Illustration src="/img/hero-illustration.png" />
  </S.Wrapper>
)

export default Main
