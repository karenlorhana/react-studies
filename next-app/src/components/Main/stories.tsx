import React from 'react'
import Main from '.'

export default {
  title: 'Main',
  component: Main
}

export const Basic = () => <Main />
