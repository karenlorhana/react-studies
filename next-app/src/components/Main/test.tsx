import React from 'react'
import { render, screen } from '@testing-library/react'
import 'jest-styled-components'

import Main from '.'

describe('<Main/>', () => {
  it('should render the heading', () => {
    const { container } = render(<Main />)

    expect(
      screen.getByRole('heading', { name: /React AvanÃ§ado/i })
    ).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render the colors correctly', () => {
    const { container } = render(<Main />)

    expect(container.firstChild).toHaveStyle({ 'background-color': '#ffffff' }) //toHaveStyle verifica os estilos
  })
})
