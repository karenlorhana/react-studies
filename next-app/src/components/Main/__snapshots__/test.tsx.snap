// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main/> should render the heading 1`] = `
.c0 {
  background-color: #ffffff;
  color: #060920;
  width: 100%;
  height: 100%;
  padding: 3rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  text-align: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c1 {
  width: 170px;
  height: 170px;
  margin-bottom: 2rem;
}

.c2 {
  font-size: 2.5rem;
}

.c3 {
  font-size: 2rem;
  font-weight: 400;
}

.c4 {
  width: 450px;
  height: 300px;
  margin-top: 2rem;
  width: -webkit-min-content(30rem,100%);
  width: -moz-min-content(30rem,100%);
  width: min-content(30rem,100%);
}

<main
  class="c0"
>
  <img
    alt="Logo do react"
    class="c1"
    src="/img/react-image.png"
  />
  <h1
    class="c2"
  >
     React Avan√ßado
  </h1>
  <h2
    class="c3"
  >
    Typescript, ReactJS, NextJS and Styled Components
  </h2>
  <img
    class="c4"
    src="/img/hero-illustration.png"
  />
</main>
`;
